File: src/main/java/com/footverse/auth/dto/PasswordResetTokenResponse.java
package com.footverse.auth.dto;

/**
 * Result of a successful OTP verification (dto-spec §6).
 *
 * @param resetToken the opaque single-use token, consumed by {@link ResetPasswordRequest}
 * @param expiresIn  the configured reset-token TTL in seconds
 */
public record PasswordResetTokenResponse(String resetToken, Long expiresIn) {
}

File: app/lib/features/auth/models/password_reset_token_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'password_reset_token_response.g.dart';

/// Result of a successful OTP verification (dto-spec §6).
@JsonSerializable(createToJson: false)
class PasswordResetTokenResponse {
  const PasswordResetTokenResponse({
    required this.resetToken,
    required this.expiresIn,
  });

  factory PasswordResetTokenResponse.fromJson(Map<String, dynamic> json) =>
      _$PasswordResetTokenResponseFromJson(json);

  final String resetToken;
  final int expiresIn;
}
